mw/i2c_service/controller/pca9685/controller.cpp

```diff
@@ -84,23 +84,20 @@ core::ErrorCode PCA9685::AutoSetServoPosition(const uint8_t &actuator_id, const
         pac_logger_.LogWarn() << ("Not find service");
         return core::ErrorCode::kNotDefine;
     }
-    if (it->second.position != state) {
-        it->second.position = state;
-        if (it->second.need_mosfet) {
-            pac_logger_.LogError() << "turn on servo xD";
-            std::ignore = std::async(std::launch::async, &PCA9685::MosfetFunc,
-                                                        this, it->second.mosfet_id, it->second.mosfet_time);
-        }
-        std::this_thread::sleep_for(std::chrono::milliseconds(it->second.servo_delay));
-        this->SetServo(it->second.channel, it->second.position == 1 ? it->second.on_pos : it->second.off_pos);
-        if (it->second.need_loosening) {
-            std::this_thread::sleep_for(std::chrono::milliseconds(LOSENING_DEFAULT_DELAY));
-            this->SetServo(it->second.channel,
-                        it->second.position == 1 ? it->second.on_loosening : it->second.off_loosening);
-        }
-        return core::ErrorCode::kOk;
+    it->second.position = state;
+    if (it->second.need_mosfet) {
+        pac_logger_.LogError() << "change servo position to " << state;
+        std::ignore = std::async(std::launch::async, &PCA9685::MosfetFunc,
+                                                    this, it->second.mosfet_id, it->second.mosfet_time);
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(it->second.servo_delay));
+    this->SetServo(it->second.channel, (it->second.position == 1) ? it->second.on_pos : it->second.off_pos);
+    if (it->second.need_loosening) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(LOSENING_DEFAULT_DELAY));
+        this->SetServo(it->second.channel,
+                    (it->second.position == 1) ? it->second.on_loosening : it->second.off_loosening);
```

Comment:
może to wszystko zamknąć w jednej funkcji i bez async ?

Reply:
mozna wsm, tylko tu timingi chcialem zachowac, w teori powinny byc idealne dzieki temu